docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
+
+    Json::Value device_dictionary;
+    device_dictionary["name"] = tcp_device->name();
+    device_dictionary["address"] = tcp_device->in_addr();
+
+    Json::Value applications_dictionary;
+    for (ApplicationHandle app_handle : tcp_device->GetApplicationList()) {
+      if (FindEstablishedConnection(tcp_device->unique_device_id(),
+                                    app_handle)) {
+        int port = tcp_device->GetApplicationPort(app_handle);
+        if (port != -1) {  // don't want to store incoming applications
+          applications_dictionary.append(std::string(port));
+        }
+      }
+    }
 
-```
-#include "transport_manager.h"
-#include "transport_manager_default.h"
+    if (!applications_dictionary.empty()) {
+      device_dictionary["applications"] = applications_dictionary;
+      devices_dictionary.append(device_dictionary);
+    }
+  }
 
-class MyTransportManagerListenerImpl: public TransportManagerListener
-{
-    // Implement interface
-}
-{
-    TransportManager *tm_impl = TransportManagerDefault::instance();
-    TransportManagerListener *my_tm_listener = new MyTransportManagerListenerImpl();
-    tm_impl->addEventListener(my_tm_listener)
+  Json::Value& dict = last_state().get_dictionary();
+  dict["TransportManager"]["TcpAdapter"]["devices"] = devices_dictionary;
 }
 ```
 
-## Adding Custom Transport Adapters to the Default TM
-
-!!! note
-
-About Transport Adapters in general:
-
-  * TM may contain zero to N Transport Adapters (a TM with 0 TAs does not do anything)
-  * Each adapter corresponds to a specific type of transport (Bluetooth, USB, TCP, etc.)
-  * TA implements transport specific search, connect, disconnect, and data transfer routines
-
-!!!
-
-!!! note
-
-Several instances of TA:
-
-  * Several TAs of the same type can exist in the TM
-  * The results of using several instances of the default TAs is not defined
-  * It is the developer's responsibility to create a custom Adapter that operates well under these conditions
-
-!!!
-
-!!! note
-
-Custom TA
-
-The simplest way to add a custom TA is to derive it from the existing implementation of the TCP or BT adapter and add it to the TM
-
-!!!
-
-```
-#include "transport_manager.h"
-#include "transport_manager_default.h"
-
-{
-    // The default implementation of TCP adapter is used in this example
-    // The developer is free to create their own implementation
-    TransportAdapter *my_transport_adapter = static_cast<TransportAdapter*>(new TcpTransportAdapter);
-    my_transport_adapter->init(); // note: TA must be initialized before getting an instance of TM
-    TransportManager *tm_impl = TransportManagerDefault::instance(); // note: When custom TA is added to the TM, the default listener will be assigned to it automatically
-    tm_impl->addTransportAdapter(my_transport_adapter)
-}
-
-```
+#### Transport Adapter Structure
 
-### Initialization
+|||
+Figure 3: Transport Adapter UML Diagram
+![TA](./assets/ta.png)
```

Comment:
There is a lot going on in this picture. Is this a inheritance diagram or showing communication between components or both? I think the dotted lines mean the component inherits from that interface, while the solid lines show ownership of that class as a class property? 

Image could use some prefacing

Reply:
I will put the same UML infos next to this diagram and again ensure this picture is not causing more confusion than it's worth.